rtCommon/bidsArchive.py

```diff
@@ -654,3 +723,89 @@ def getIncremental(self, imageIndex: int = 0, **entities) \
         except MissingMetadataError as e:
             raise MissingMetadataError("Archive lacks required metadata for "
                                        "BIDS Incremental creation: " + str(e))
+
+    @failIfEmpty
+    def getBidsRun(self, **entities) -> BidsRun:
+        """
+        Get a BIDS Run from the archive.
+
+        Args:
+            entities: Entities defining a run in the archive.
+
+        Returns:
+            A BidsRun containing all the BidsIncrementals in the specified run.
+
+        Raises:
+            NoMatchError: If the entities don't match any runs in the archive.
+            QueryError: If the entities match more than one run in the archive.
+
+        Examples:
+            >>> archive = BidsArchive('/tmp/dataset')
+            >>> run = archive.getBidsRun(subject='01', session='02',
+                                         task='test-task', run=1)
+            >>> print(run.numIncrementals())
+            53
+        """
+        images = self.getImages(**entities)
+        if len(images) == 0:
+            raise NoMatchError(f"Found no runs matching entities {entities}")
+        if len(images) > 1:
+            entities = [img.get_entities() for img in images]
+            raise QueryError("Provided entities were not unique to one run; "
+                             "try specifying more entities "
+                             f" (got runs with these entities: {entities}")
+        else:
+            bidsImage = images[0]
+            niftiImage = bidsImage.get_image()
+            niftiData = getNiftiData(niftiImage)
+            metadata = self.getSidecarMetadata(bidsImage)
+            metadata.pop('extension')  # only used in PyBids
+
+            run = BidsRun(**bidsImage.get_entities())
+            for imageIdx in range(niftiImage.header.get_data_shape()[3]):
+                newData = niftiData[..., imageIdx]
+                newImage = nib.Nifti1Image(newData,
+                                           niftiImage.affine,
+                                           niftiImage.header)
+
+                newIncremental = BidsIncremental(newImage, metadata)
```

Comment:
Probably should have asked this on the previous review, but I could envision the BidsRun just keeping the niftiData array and then bidsRun.getIncremental() doing the work of peeling off the correct volume (rather than creating each incremental ahead of time). What do you think are the pros/cons?

Reply:
Finding essentially only pros, I've gone ahead and made the change.